### PR TITLE
Add/35173 category field improvements

### LINKED ISSUE #35173: Category management improvements
**Description:** the new category management flow has some flaws that should be fixed before we test the new experience with users

**Acceptance criteria:**

- [x] The user should see an option to create a new category when he starts typing
- [x] The category list should appear on the first click

### PR DESCRIPTION
**Pull request by mdperez86**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

- Add new experimental prop to the select control to externalize the functionality of open the dropdown menu when clicking the select control.
- Use the prev prop in the category field to open the menu when click the input field.

Closes https://github.com/woocommerce/woocommerce/issues/35173

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Go the Add New (MVP) product page
2. Click on Categories field
3. A dropdown menu with the category list should be shown

<video src="https://user-images.githubusercontent.com/13334210/202287612-aaa2c7dd-9390-4e26-a116-67bc40ad0bed.mov"></video>

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/components/src/experimental-select-control/select-control.tsx [hunk 1]
```diff
@@ -63,6 +63,15 @@ type SelectControlProps< ItemType > = {
 	selected: ItemType | ItemType[] | null;
 	className?: string;
 	disabled?: boolean;
+	/**
+	 * This is a feature already implemented in downshift@7.0.0 through the
```
Comment (joshuatf) [inline packages/js/components/src/experimental-select-control/select-control.tsx:67]: Thanks for this comment 💯

File: plugins/woocommerce-admin/client/products/fields/category-field/category-field.tsx [hunk 2]
```diff
@@ -205,14 +206,13 @@ export const CategoryField: React.FC< CategoryFieldProps > = ( {
 							className="woocommerce-category-field-dropdown__menu"
 						>
 							<>
-								{ isOpen && isSearching && items.length === 0 && (
+								{ isSearching ? (
```
Comment (joshuatf) [inline plugins/woocommerce-admin/client/products/fields/category-field/category-field.tsx:209]: @louwie17 You had a similar logic change in a recent PR IIRC.  Just want to ping you to make sure there aren't any conflicts.

File: packages/js/components/src/experimental-select-control/select-control.tsx [hunk 3]
```diff
@@ -249,6 +259,7 @@ function SelectControl< ItemType = DefaultItemType >( {
 						onFocus( { inputValue } );
 					},
 					onBlur: () => setIsFocused( false ),
+					onClick: () => __experimentalOpenMenuOnClick && openMenu(),
```
Comment (joshuatf) [inline packages/js/components/src/experimental-select-control/select-control.tsx]: One quirk is that when we click around the combobox area that's not the input (e.g., click on the magnifying glass icon) it does not open the menu.

Should we add this logic `__experimentalOpenMenuOnClick && openMenu()` here as well?  https://github.com/woocommerce/woocommerce/blob/e924344b8594493d7d4ddbf2dae4592f9cdefa24/packages/js/components/src/experimental-select-control/combo-box.tsx#L32-L35

**Review by joshuatf — DISMISSED** [review]
Nice workaround and awesome job finding the downshift updates!

This is testing well for me.  Left one possible addition, but pre-approving ✅

File: plugins/woocommerce-admin/client/products/fields/category-field/category-field.tsx (hunk 2 above)
Comment (louwie17) [reply plugins/woocommerce-admin/client/products/fields/category-field/category-field.tsx:209]: I do believe this logic update is fine and shouldn't cause any conflicts

**Review by joshuatf — DISMISSED** [review]
Looks good with focus now!  That also solved the combobox click issue mentioned above.

There are some failing tests, but I think unrelated to this PR.  I restarted them to try and fix.  Otherwise this LGTM 🚢

**Review by joshuatf — APPROVED** [review]
LGTM :shipit:

**Pull request merged**